### DISCUSSION TITLE
Drun Error Message

### DISCUSSION THREAD

**Discussion by kvnduff** [davatorium/rofi#1630] [Ideas/Feature Requests]
I'm wondering if there is a Rofi config option to change the following behavior? If not, I wonder if a configuration option might be considered:

Behavior described:
1. Activate drun to open a .desktop application
2. Type in a string of letters that filters out all of the listed applications
3. Type return
4. The result of 1-3 above is that the rofi window moves to the bottom of the screen (this may be a result of my own window manager) and the following message is displayed in the window (I think this is currently desired Rofi behavior):
```
Failed to execute: 'string typed into the search box'
Error: 'Failed to execute child process "string typed into search box" (No such file or directory)'
```

I'm wondering if there is a way to supresss this error message and to prevent the rofi window from moving location (if the window movement is a result of Rofi and not potentially in my case, the window manager). In short, the desired behavior would be that if a string is typed into the search box so that no applications are highlighted (i.e. all applications are filtered out), then hitting return would not move the window location, and not display an error message.

**Comment by DaveDavenport** [comment]
Sounds like a broken theme.  you need to provide way more information to say anything about this.
it is up for the theme to implement ths error dialog.

**Reply by DaveDavenport** [reply]
Can you try with:

```bash
rofi -show drun -no-config
```

Also maybe mention:
* what window manager are you running
* What rofi version are you running
* What is your configuration 
* What is your theme

**Reply by kvnduff** [reply]
I tried with rofi -show drun -no-config and the behavior was what you demonstrated in your screen capture. The window location isn't too important, but I think supressing the error message entirely might be nice to make the interface a bit cleaner. The information you requested is as follows:
- window manager: AwesomeWM version 4.3
- rofi version: 1.7.3
- my config and theme are attached as text files (couldn't upload .rasi using the file picker)
[config.txt](https://github.com/davatorium/rofi/files/8642175/config.txt)
[myTheme.txt](https://github.com/davatorium/rofi/files/8642176/myTheme.txt)

**Reply by DaveDavenport** [reply]
I got two major issues with adding this feature:
1. you use-case does not make sense to me.  You want to type in 'rubbish' into the entry box, getting clearly a visual indicator that there is no match. Then still go through the process  try to launch the data you entered, but then don´t want to get notified it failed to launch it?   
2. Hiding errors when the tools fails to perform the users request, especially that is the *key* feature of the current tool is bad UI design. This will be very confusing to the user.

**Reply by kvnduff** [reply]
Ok, that makes sense. At the risk of sounding argumentative or unappreciative (which I definitely am not, I love Rofi!), I would like to explain my use case, which may or may not apply to other users. When I open the rofi drun menu, I usually quickly type a few characters and press enter before actually waiting to see what application has been selected. I realize this approach could be considered reckless. Sometimes I make a mistake and type a string that matches an application I didn't intend to open or I type a string that doesn't match any applications which results in the error dialogue. So yes, sometimes I type 'rubbish' and still type enter even though I could have waited to check which application (or no application) was visually selected. I'm not sure but I imagine many users use the drun menu in this manner to save time rather than to be accurate. However, in my opinion the error message is unecessary as you could press enter, and if nothing opens, then you would see that there are no applications selected. That siad, I understand the points you made and respect the approach taken. Again, thank you for this project and the time you have devoted to it! :)

**Reply by DaveDavenport** [reply]
I think what is missing here in the understanding, if there is no match, rofi tries to execute the input as an application executable name.
A lot of users use this to launch programs that do not have a desktop file. As proof, I got many complaints when this was once disabled. (look back in the issue tracker).

**Reply by kvnduff** [reply]
Ahhh, that makes sense! Thanks so much for your replies and my appologies for taking up your time.

**Reply by DaveDavenport** [reply]
No problem.

**Reply by kvnduff** [reply]
Thanks again! And just in case someone else that isn't too technically competent (like myself) comes across this thread, I needed to add "anchor: center" to my theme so that the error message would stay centered in the screen (insteasd of how it originally moved below the regular window position).

**Comment by DaveDavenport** [comment]
![drun-error](https://user-images.githubusercontent.com/406245/167169489-f374fe23-636e-4ab4-9000-e3c74e48cde8.gif)

**Reply by DaveDavenport** [reply]
here the error dialogs centers, as that is how the config is setup.

**Reply by kvnduff** [reply]
Sorry for not providing the required information and thanks for your quick reply and for the video recording to demonstrate default Rofi behavior. I don't want to waste your time with this discussion/request but just wondering, besides, the window location, are there any configuration options to supress the error message so that nothing happens if you type return when no applications are highlighted? This is really not an important feature so if this isn't possible then it's probably not something worth considering. I'm just wondering if this is something that can currently be modified. Sorry if I've missed something in the documentation but I couldn't find anything about it. Thanks for your work maintaining Rofi - it's an amazing project!

**Reply by DaveDavenport** [reply]
I think I found a reason why (if not set explicitly) it might position it with anchor set at north. I will fix this in git.
